homeassistant/components/aemet/__init__.py

```diff
@@ -57,3 +61,11 @@ async def async_update_options(hass: HomeAssistant, entry: ConfigEntry) -> None:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Remove a config entry."""
+    await hass.async_add_executor_job(
+        shutil.rmtree,
+        hass.config.path(STORAGE_DIR, "aemet"),
+    )
```

Comment:
How does this work if you have 2 entries?

Reply:
> How does this work if you have 2 entries?

@joostlek the API call of "maestro/municipios" will be the same for both entries, creating the following file:
`maestro-municipios.json`

Each entry will create the following files for the hour/day forecasts:
```
prediccion-especifica-municipio-diaria-${TOWN_ID}.json
prediccion-especifica-municipio-horaria-${TOWN_ID}.json
```
If TOWN_ID is the same, then both will r/w the same files.

The same behaviour applies for the station data, if both entries rely on the same weather station, and the station data is enabled, then both entries will r/w the same file for the station.

However, I don't think that any user has more than 1 entry anyway...

Reply:
It's quite a common thing that users have 2 entries. For example for either work, or places they are often.

But I'm more curious about how the deleting works here, since it deletes the full cache and not just a single file. Can the other entry handle that?

Reply:
@joostlek excuse me but I didn't get your point at first.
It should be fixed by 793b4bdef110cc87e7aba073a49e06f116dda6ed since the Unique ID is now added to the cache directory name.

Reply:
Awesome 🚀